Dockerfile

```diff
@@ -28,18 +27,26 @@ WORKDIR /var/lib/hypothesis
 
 # Copy packaging
 COPY h/__init__.py h/_version.py ./h/
-COPY README.rst package.json setup.* requirements.txt ./
+COPY README.rst setup.* requirements.txt ./
+COPY scripts ./scripts/
+COPY package.json gulpfile.js ./
 
 # Install application dependencies.
-RUN npm install --production \
-  && pip install --no-cache-dir -r requirements.txt \
-  && npm cache clean
+RUN pip install --no-cache-dir -r requirements.txt
 
 # Copy the rest of the application files
 COPY gunicorn.conf.py ./
 COPY conf ./conf/
 COPY h ./h/
-COPY scripts ./scripts/
+
+# Build client and static assets
+
+## Remove dependencies required only for unit testing
+RUN sed -i -r '/karma|proxyquire|phantom|mocha|chai|jscs/d' package.json
```

Comment:
I wonder if everything needed to build the docker image could go in dependencies, and then only the things being removed by sed here could go in devDependencies, then you could just use `npm install --production` instead of 'sed`?

Since our only officially supported "production" deployment is the docker one, it might make sense that `npm install --production` is for use in the Dockerfile.